tests/components/stream/test_worker.py

```diff
@@ -209,7 +210,8 @@ async def test_stream_open_fails(hass):
     stream.hls_output()
     with patch("av.open") as av_open:
         av_open.side_effect = av.error.InvalidDataError(-2, "error")
-        stream_worker(STREAM_SOURCE, {}, stream.outputs, threading.Event())
+        segment_buffer = SegmentBuffer(stream.outputs)
```

Comment:
Alternative idea: Creation of a `SegmentBuffer` could also be a method on `Stream` to hide/ecapsulate the creation of the outputs callback.

Reply:
Up to you, I don't think it's too messy as is (we only call it once in the code, but I guess it would be cleaner in the tests)